backend/models/donation_subscription_payment.py

```
@@ -0,0 +1,162 @@
+    if not sale_id:
+        return None
+
+    doc = await DB.db.donation_subscription_payments.find_one({"paypal_txn_id": sale_id})
+    if not doc:
+        return None
+
+    existing_refunds = _normalize_refunds(doc.get("refunds") or [])
+    if refund.refund_id and any(r.refund_id == refund.refund_id for r in existing_refunds):
+        refunds = existing_refunds
+    else:
+        refunds = existing_refunds + [refund]
```

Ruff (0.14.5)

Consider `[*existing_refunds, refund]` instead of concatenation

Replace with `[*existing_refunds, refund]`

(RUF005)